fip-6.md

```diff
@@ -0,0 +1,376 @@
+---
+fip: 6
+title: Provide locked token capability for use by foundation and FIO users
+status: Draft
+type: Functionality
+author: Ed Rotthoff <ed@dapix.io>
+created: 2020-04-16
+updated: 
+---
+
+## Abstract
+This FIP implements the following:
+* Adds configurable locking mechanism for FIO tokens which is usable by all users of FIO 
+* Locked tokens will have any number of lock periods, each period will have a duration and a percent to unlock after the duration. Durations will be processed relative to the locks creation time.
+* Users will be able to lock tokens within an account that they own (transfer is not required to lock tokens)
+* Users will be able to unlock an amount of tokens that are locked (when the lock is specified to be unlockable)
+* Users will be able to vote using locked tokens (when the lock is specified to be votable)
+* Adds a new endpoint which will support creation of locks that is called lock_tokens.
+* Adds a new endpoint which will support cleaning of the system by BPs and users that is called clean_lock_table.
+* Adds a new endpoint which will support cleaning of an accounts locks that is called clean_locks
+* Adds an endpoint which will list the locks associated with a FIO public key called get_locks
+* Modify the get_fio_balance endpoint to return the amount of locked tokens along with the balance.
+* Modify the transfer of tokens and voting to perform the update of the locks for an account based on the specified periods.
+
+## Motivation
+Presently the FIO API provides a fixed set of locked token types which have been used at FIO genesis, the types of locks and associated characteristics of locks were implemented in code without the ability to configure the locks without re-coding the locking logic. We make a new data model for a lock which consists of a set of locking periods which are relative to the time of creation of the lock, and flags relating to the lock (votable, unlockable). These new locks do not relate to the "main net" locks used to provide tokens to FIO investors and employees at FIO genesis. New locks and old locks will all be processed with equal precedence according to the rules for the lock. :
```

Comment:
I'd like to see Backwards compatibility section (per FIP template) that talks about how these locks play with the Mainnet locks. Also explain this more, I'm having problems understanding how those locks interact.

Reply:
there is no backwards compatibility, the locks don't interact, all types of locks are honored and observed.